CPS-?/README.md

```diff
@@ -0,0 +1,170 @@
+---
+CPS: ???
+Title: Cardano Governance Security
+Category: ?
+Status: Open
+Authors:
+    - Richard McCracken <rickymac68@icloud.com>
+    - Andrew Westberg <andrewwestberg@gmail.com>
+Proposed Solutions: N/A
+Discussions:
+    - https://github.com/cardano-foundation/CIPs/pull/490
+    - https://github.com/cardano-foundation/CIPs/pull/491
+Created: 2023-03-28
+---
+
+# Cardano Governance Security
+
+## Abstract
+
+As Cardano's community-led governance system goes live, it's crucial to address the security of the governance mechanism in a way that informs all participants of potential weaknesses and how to mitigate them. The guidelines provided in the CIP are relevant to various groups, including Constitutional Committee members, Delegation Representatives, Stake Pool Operators, voters, and all Ada holders. Moreover, the guidelines are applicable to wallet, dApp, and infrastructure developers to help them build situational awareness and design governance security considerations into their development processes.
+
+These recommendations are the result of a collaborative effort among cybersecurity, financial, and game theory experts, as well as participants of the Voltaire workshop, including developers, stake pool operators, representatives from IOG, Cardano Foundation, and Emurgo, members of Project Catalyst, and members of the Cardano community.
+
+
+## Problem
+
+The Cardano community's on-chain governance mechanism, Voltaire, is susceptible to various types of attacks that may compromise its security. This system defines on-chain decision-making that is automatically executed when a certain threshold of approval by human actors is met. To ensure maximum awareness of potential threats and vulnerabilities, this document will enumerate and describe each threat or vulnerability, including recommended mitigation techniques. It is crucial that all stakeholders, including voters, developers, and governance actors, are informed of these risks to identify and prevent any damage or, in the case of a threat event, facilitate recovery.
+
+In rare instances, a threat may be identified that requires responsible reporting procedures to prevent exposing an active vulnerability to bad actors seeking to exploit it. In such cases, it is essential to notify the appropriate group or entity with the ressources, or are responsible for mitigating the threat in a confidential manner.
+
+### #1: Unknown software vulnerability affecting governance is discovered.
+
+An assumption is that are no known software back doors. Unknown software vulnerabilities may be discovered.
+
+Recommendation: The person discovering the vulnerability should document the circumstances and details in which the vulnerability is present and how to reproduce the vulnerability. Then report the vulnerability in a confidential manner using a secure method to the developer team with the resources to correct the vulnerability.
+
```

Comment:
A tiered bug bounty rewards program can be employed to recompense those who articulate meaningful corrections required to reverse/repair a potential vulnerability. A treasury may need to be formed to fund this specific activity, releasing those proposed rewards by a committee vote.

Reply:
Ok Chris, I will add that to the recommendations. Be aware it is un-enforceable as the bug bounty has to be paid by some entity. Good recommendation though.

Reply:
Would it make sense to create a multisig arrangement tied to an address where people can donate for this purpose and bounties are paid out by multisig. I would personally donate 10 Ada per epoch to it.

Reply:
What is meant by "unenforceable" and "paid by some entity"?

It feels like the whole bounty payout could be managed on-chain with the payout in ada or another native token (djed perhaps).

I haven't thought through this idea fully yet so be gentle...It feels like the whole flow could (maybe not should?) be managed on-chain.
"Responsible" party submits encrypted vulnerability details inside the metadata of [a] transaction(s).  Provides the decryption key to the "relevant parties" as well as payment address for any bounty (payment addr could be included in metadata). Once bug is resolved and paid the encryption key for the vuln data can be released so the public can view details

How much the bounty is worth will need to be determined somehow? Who votes on that? Going to be difficult without the public knowing the vuln details.  Feels like we need a separate "treasury pool" specifically for bug bounties that can be sined/paid by a subset of the governance members or a new class of dreps?